### PR TITLE
[#13] Migrate to Swift 5

### PR DESCRIPTION
**Pull request by kam800**

### CODE REVIEW

File: MachObfuscator/Commons/Pipe.swift [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+precedencegroup ForwardApplication {
+    associativity: left
+}
+
+infix operator |>: ForwardApplication
+
+func |> <T, U>(value: T, function: (T) -> U) -> U {
+    return function(value)
+}
```
Comment (kkosowski-bt) [inline MachObfuscator/Commons/Pipe.swift]: Perhaps we could also add function composition operator:

```
precedencegroup ForwardComposable {
    associativity: left
    higherThan: ForwardApplicable
}

infix operator >>>: ForwardComposable

public func >>><A, B, C>(_ f: @escaping (A) -> B, _ g: @escaping (B) -> C) -> (A) -> C {
    return {
        g(f($0))
    }
}

```

File: MachObfuscator/DataAccess/UnsafeRawPointer+Structs.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+extension UnsafeRawPointer {
+    func getStruct<T>() -> T {
+        return bindMemory(to: T.self, capacity: 1)
+            .pointee
+    }
+
+    func getStructs<T>(count: Int) -> [T] {
+        return bindMemory(to: T.self, capacity: count)
+            |> { UnsafeBufferPointer<T>(start: $0, count: count) }
+            |> [T].init
```
Comment (kkosowski-bt) [inline MachObfuscator/DataAccess/UnsafeRawPointer+Structs.swift]: The latter two could be composed with `>>>` operator :)

File: MachObfuscator/Commons/Pipe.swift (hunk 1 above)
Comment (kam800) [reply MachObfuscator/Commons/Pipe.swift]: Good idea. What to you think about adding this in another PR? After all this should be just migration to Swift 5 😂

Comment (kkosowski-bt) [reply MachObfuscator/Commons/Pipe.swift]: Haha right! Got overexcited!

**Pull request merged**